### ISSUE TITLE
Code example in root Readme doesn't work

### ISSUE THREAD

**Issue by echeran** [unicode-org/icu4x#2352]
@sven-oly was trying to copy-paste the example code in root `README.md`, and together we couldn't get it to work.  Plus, it's using an old version number (`0.2`).

We compared it to [the example in the root of the `icu` component](https://unicode-org.github.io/icu4x-docs/doc/icu/index.html#example), which was similar yet noticeably different.

**Comment by sffc** [comment]
Yeah, don't trust any code samples that are not from docs.rs until we have everything updated (which should happen in advance of the 1.0 release)
